### PR TITLE
Custom test client

### PR DESCRIPTION
**Pull request by singingwolfboy**
Having a custom test client makes it easier to write automated tests using Flask-Login.

### CODE REVIEW

**Comment by maxcountryman** [comment]
This makes sense to me! 👍

**Comment by singingwolfboy** [comment]
Great! Do you have any feedback on the functionality or the documentation? I know that Flask-Login sets more values in the session cookie than `user_id` and `_fresh`, but those two were the easiest to handle, so I only included those. If you have a suggestion for how to handle the other values, I'd love to hear it! Alternatively, we can start with minimal functionality for now, and add more over time.

And are the docs clear? I always want to verify that my documentation is understandable to someone who _didn't_ write the code. 😄

File: flask_login/test_client.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from flask.testing import FlaskClient
+
+
+class FlaskLoginClient(FlaskClient):
+    """
+    A Flask test client that knows how to log in users
+    using the Flask-Login extension.
+    """
+
+    def __init__(self, *args, **kwargs):
```
Comment (alanhamlett) [inline flask_login/test_client.py]: This way is more explicit:
`def __init__(self, user=None, fresh_login=True, *args, **kwargs):`

I know the `FlaskClient` docs show using `kwargs.pop` so either is fine.

**Comment by singingwolfboy** [comment]
Great, thanks for merging! Any idea when there will be a new release of Flask-Login, so people can use this feature?

**Comment by alanhamlett** [comment]
@maxcountryman does the releases. Thanks for adding this, it will be very helpful when testing.

**Comment by singingwolfboy** [comment]
Cool. Let's move this discussion over to issue #425.

**Pull request merged**